pallets/confidential-asset/src/lib.rs

```diff
@@ -680,6 +682,82 @@ impl<T: Trait> Module<T> {
     ) {
         TxPendingState::remove((owner_did, account_id, instruction_id));
     }
+
+    /// Transfers an asset from one identity's portfolio to another.
+    pub fn base_confidential_transfer(
+        from_did: &IdentityId,
+        from_account_id: &MercatAccountId,
+        to_did: &IdentityId,
+        to_account_id: &MercatAccountId,
+        tx_data: &JustifiedTransferTx,
+        instruction_id: u64,
+    ) -> DispatchResult {
+        // Read the mercat_accounts from the confidential-asset pallet.
+        let from_mercat_pending_state =
+            Self::mercat_tx_pending_state((from_did, from_account_id, instruction_id))
+                .to_mercat::<T>()?;
+
+        // Get receiver's account.
+        let to_mercat = Self::mercat_accounts(to_did, to_account_id).to_mercat::<T>()?;
+
+        // Get sender's account.
+        let from_mercat = Self::mercat_accounts(from_did, from_account_id).to_mercat::<T>()?;
+
+        // Verify the proofs.
+        let mut rng = rng::Rng::default();
+        let _result = TransactionValidator
+            .verify_transaction(
+                tx_data,
+                &from_mercat,
+                &from_mercat_pending_state,
+                &to_mercat,
+                &[],
+                &mut rng,
+            )
+            .map_err(|_| {
+                // Upon transaction validation failure, update the failed outgoing accumulator.
+                let _result = Self::add_failed_outgoing_balance(
+                    from_did,
+                    from_account_id,
+                    tx_data
+                        .finalized_data
+                        .init_data
+                        .memo
+                        .enc_amount_using_sender,
+                );
```

Comment:
Did you mean to use the result as the name `_result` suggested?
```suggestion
                ).unwrap();
```
If not then renaming the variable to `_` should clear the confusion.

Reply:
I don't think I should use `unwrap()` since that'd make the code panic in case of error. I'll ignore the result since it's throwing an error any way.